pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -208,6 +208,26 @@ func (s *Service) RegisterInstanceWithAPIServerELB(i *infrav1.Instance) error {
 	return nil
 }
 
+// DeregisterInstanceFromAPIServerELB de-registers an instance from a classic ELB
+func (s *Service) DeregisterInstanceFromAPIServerELB(i *infrav1.Instance) error {
+	name, err := GenerateELBName(s.scope.Name())
+	if err != nil {
+		return err
+	}
+
+	input := &elb.DeregisterInstancesFromLoadBalancerInput{
+		Instances:        []*elb.Instance{{InstanceId: aws.String(i.ID)}},
+		LoadBalancerName: aws.String(name),
+	}
+
+	_, err = s.scope.ELB.DeregisterInstancesFromLoadBalancer(input)
+	if err != nil {
```

Comment:
This should probably differentiate a "NotFound" error from other errors, since this is one of the first things called in `reconcileDelete` and should be re-entrant.

Reply:
The API is not throwing error in case we are registering an instance that is already de-registered, so I think we are already ok.